### ISSUE TITLE
[C++ Headers] Non-sensical auto-generated if-statement leads to compilation error when compiled with `-Werror`

### ISSUE THREAD

**Issue by aentinger** [OpenCyphal/nunavut#271]
As the compiler says, a uint8_t can not have an larger value than 255, consequently the if statement is dead code.

**Compiler error**
```bash
reg/udral/service/battery/Status_0_2.hpp:462:26: error: comparison is always false due to limited range of data type [-Werror=type-limits]
  462 |             if ( _size0_ > 255U)
```

**reg/udral/service/battery/Status_0_2.hpp**
```C++
        const std::uint8_t _size0_ = in_buffer.getU8(8U);
        in_buffer.add_offset(8U);
            if ( _size0_ > 255U)
```

**Comment by aentinger** [comment]
I'm only partially convinced. If OpenCyphal targets reliable systems its to be expected that whole codebase (including auto-generated code) does not break when using `-Werror`. This error may be trivial, but others may be not.

**Comment by pavel-kirienko** [comment]
The matter of `-Werror` is orthogonal to the integrity of the codebase. Compilers are known to occasionally produce false diagnostics, which are promoted to build-breaking errors when strict compilation settings are given. This causes the compiler to reject valid code. It is, generally speaking, impossible to produce any non-trivial code that is certain to emit no diagnostics from all existing standard-compliant compilers. It follows that:

1. `-Werror` and similar strict options should only be used with the code you control directly and never be applied to third-party code.
2. Build error with `-Werror` does not indicate that the code is invalid or otherwise unacceptable.

**Comment by thirtytwobits** [comment]
I'm inclined to accept this bug. I don't like normalizing deviance and if we can generate code that avoid well-known compiler warnings we should.

**Comment by thirtytwobits** [comment]
This should be fixed with 2.0.7
